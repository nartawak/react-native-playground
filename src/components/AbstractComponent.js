import { Component } from 'react';

export default class AbstractComponent extends Component {
  /**
   * Constructor
   * @param props
   */
  constructor(props) {
    super(props);

    if (this.constructor === AbstractComponent) {
      throw new TypeError('Can not construct abstract class.');
    }
  }
}
