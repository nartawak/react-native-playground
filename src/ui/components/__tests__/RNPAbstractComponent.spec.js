import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AbstractComponent from '../RNPAbstractComponent';

/**
 * Custom component that extends AbstractComponent
 * @class ExtendsComponent
 */
class ExtendsComponent extends AbstractComponent {}

describe('Abstract component', () => {
  it('Should not create AbstractComponent', () => {
    const createAbstractComponent = () => {
      renderer.create(
        <AbstractComponent />
      ).toJSON();
    };
    expect(createAbstractComponent).toThrowError('Can not construct abstract class.');
  });

  it('Should create ExtendsComponent', () => {
    const createAbstractComponent = () => {
      renderer.create(
        <ExtendsComponent />
      ).toJSON();
    };
    expect(createAbstractComponent).toMatchSnapshot();
  });

  it('Should get the application theme', () => {
    expect(AbstractComponent.getTheme()).toBeDefined();
    expect(AbstractComponent.getTheme().COLORS).toBeDefined();
    expect(AbstractComponent.getTheme().COLORS.BACKGROUND_COLOR).toEqual('#F5FCFF');
  });
  
  it('Should get the device width', () => {
    expect(AbstractComponent.getDeviceWidth()).toEqual(375);
  });
  
  it('Should get the device height', () => {
    expect(AbstractComponent.getDeviceHeight()).toEqual(667);
  });
});
