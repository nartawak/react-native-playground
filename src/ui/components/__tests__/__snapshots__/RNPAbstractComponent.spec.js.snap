exports[`Abstract component Should create ExtendsComponent 1`] = `[Function]`;
